internal/rest/resources/sql.go

```diff
@@ -103,7 +103,7 @@ func sqlSelect(ctx context.Context, tx *sql.Tx, query string, result *types.SQLR
 		return fmt.Errorf("Failed to execute query: %w", err)
 	}
 
-	defer rows.Close()
```

Comment:
Would it makes sense to also log an error like in https://github.com/canonical/microcluster/pull/131/commits/de53cdc8a7143bad2a468836c272701a79c6a87a? 

There were some more commits before that added `//nolint:errcheck` instead of logging an error.

Reply:
Yep can do.

Reply:
I've logged all deferred errors except one that is in a test case (where I've added a comment explaining why it is ignored with a lint rule).